### DISCUSSION TITLE
remote exit code

### DISCUSSION THREAD

**Discussion by dev72** [iapyeh/sshscript#5] [General]
Hello,
is it possible to obtain the exit code of a remote executed program?

example command:
$test -e /non/existent/file

this does not output anything on stdout nor stderr, but returns an exit code 1.

How can I access this code?

Thank you
Gabriel.

**Comment by iapyeh** [comment]
In brief, at this moment(v1.1.12), the answer is no.
I'd like to add this feature in the next release.
Temporarily, two dollars commands ($$) might be a workaround, like this:
```

existing = '/var/log/syslog'
nonexisting = '/non/existing'

# comment-out the next line to test on localhost (subprocess)
$.connect('user@host')

$$[ -e @{nonexisting} ] && echo existing || echo nonexisting
print(nonexisting, $.stdout.strip())

$$[ -e @{existing} ] && echo existing || echo nonexisting
print(existing,$.stdout.strip())

$$'''
test -e @{nonexisting}
echo @{nonexisting} exit code is $?
test -e @{existing}
echo @{existing} exit code is $?
'''
print($.stdout.strip())
```

**Comment by dev72** [comment]
ok, thank you.

**Comment by iapyeh** [comment]
Hi Gabriel, exit code is supported in v1.1.14. Thanks for your suggestion. Example:
```
$[ -e /non-existing ]
assert $.exitcode == 1
```
The exitcode is working only under some scenario, please see the table at the bottom of release note of v1.1.14 for details.
[Here is the release note of v1.1.14](https://iapyeh.github.io/sshscript/release-v1.1.14)

**Comment by dev72** [comment]
Hi, ok, thank you very much!
